src/cargo.rs

```diff
@@ -70,8 +70,15 @@ impl<'a> Cmd<'a> {
     /// Includes the type of command (e.g `test`, `run`) and the default
     /// arguments (`DEFAULT_CARGO_ARGS`).
     fn args(&self) -> impl Iterator<Item = &str> + Clone {
+        let extra_args = match self.kind {
```

Comment:
Given that it is only one flat and the current code would need to look quite different to support other `extra_args` I would rename this to `no_run_flag`. Furthermore, it would also be great to have a check that that flag is not already set in `self.args` (the field). Lastly, I think we should merge this function with `args_str`. Both are very short and the current version is a bit funky given that `self.args` is a function and a field (TIL, this is possible :D)

Reply:
Yeah I kinda threw this together and didn't really find a good name for it. `no_run_flag` is a lot better!

Regarding `args` and `args_str`, I made the latter as a convenience to prevent having to do `self.args().collect::<Vec<_>>().join(" ")` whenever we need to print the command we ran. Then I decided to try to prevent making a vec and rather make a string directly, hence the complexity of it now is remarkable. We do need a way to get an iterator of arguments for `Command`, but we could remove `args_str` and replace it with what I mentioned above.

Since the field is private there is no problem (as you learnt). It's so much nicer than having to put a prefix somewhere like `get_args` or using `m_args` for the data field. Naming is hard, so I wanna name as little as possible. 😊